### DISCUSSION TITLE
Notifications from Home Assistant sometimes don't include sub_label even if it's present

### DISCUSSION THREAD

**Discussion by sjkjs** [blakeblackshear/frigate#6105] [Ask A Question]
I'm running Frigate along with the official Home Assistant plugin, and Double Take for facial recognition. Double Take will attempt to identify faces in the images from Frigate, and then if someone is detected, it will push a sub label with the person's name back to Frigate via the API.

When I get a notification about a person appearing on a camera, it will initially have the string "A Person was detected on the ... camera", which is correct because Double Take is still waiting for its processing to complete. After Double Take has finished and pushed the sub_label back, sometimes the notification from Home Assistant  will update to say "<Person's name> was detected on the ... camera", but sometimes it stays as "A person was detected".

The sub label correctly shows under Frigate events (e.g. Person: <Name>90%) even in the cases where the notification doesn't update.

I suspect, but am not 100% sure, that it only updates the notification if a better image is found. If the sub label is present before the better image occurs, the sub label will also be added to the notification. So, if the best image occurs right at the beginning of the event then the sub label which is added later on never gets looked at.

Is it possible to fix this so that it will update the notification if a sub label is added but there is no change to the best image? Or if I'm wrong about the cause, is there another way to fix this?

**Comment by NickM-27** [comment]
MQTT messages are only sent in certain cases after the initial message:
1. Better image
2. Object changes zones
3. Stationary object every 60 seconds
4. Event ends

It would not be easy to make the notification update the MQTT sub label message considering that those are two entirely different processes.

**Reply by sjkjs** [reply]
Does that mean that as long as the sub label exists before the event ends, it should be adding the sub label to the updated notification? I don't believe that's happening in my case. Double Take only takes a couple of seconds to identify a face and push the sub label to Frigate, but sometimes even in 20+ second long events it will fail to add the sub label string to the notification.

**Reply by NickM-27** [reply]
I have a similar setup, at the end of the event if doubletake found a name it is always set in the end message in my usage
